### PR TITLE
content/guides/integrations: add Memcached Go guide

### PR DESCRIPTION
**Pull request by odeke-em**
Produced a guide to use the now instrumented Memcache
client "gomemache" forked and instrumented by Orijtech
but originally from Brad Fitzpatrick.

### CODE REVIEW

File: content/guides/integrations/memcached/Go.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+With Memcached now installed and running, we can now start the code sample and for simplicitly examining metrics, we'll use Stackdriver Monitoring and Tracing
+
+{{% notice tip %}}
+For assistance setting up Stackdriver, please visit this [Stackdriver setup guided codelab](/codelabs/stackdriver)
+{{% /notice %}}
+
+Our sample is an application excerpt from a distributed prime factorization engine that needs to calculate square roots
+of big numbers but would like to reuse expensively calculated results since square roots of such numbers are CPU intensive.
+To share/memoize results amongst our distributed applications, we'll use Memcache. In the first round, before a cache hit, we'll
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply content/guides/integrations/memcached/Go.md]: Memoize is a different word and used in the context of CS https://en.wikipedia.org/wiki/Memoization

File: content/guides/integrations/memcached/Go.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+	log.SetFlags(0)
+
+	cst := httptest.NewServer(&ochttp.Handler{
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			qv := r.URL.Query()
+			query := qv.Get("v")
+			ctx := r.Context()
+
+			// Check Memcached if we've computed it before
+			memoizedSQRT, err := mc.Get(ctx, query)
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply content/guides/integrations/memcached/Go.md]: Please see the comment above about `memoize` or https://en.wikipedia.org/wiki/Memoization

File: content/guides/integrations/memcached/Go.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+##### Available metrics
+
+* Also a couple of metrics have been added:
+
+Metric|Name|Description
+---|---|---
+Number of Cache Misses|`cache_misses`|The number of cache misses
+Number of Cache Hits|`cache_hits`|The number of cache hits
+Number of Client errors|`client_errors`|The number of general client errors
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply content/guides/integrations/memcached/Go.md]: Roger that, thanks!

File: content/guides/integrations/memcached/Go.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+all now take in a context.Context object as the first argument, to allow for trace propagation.
+
+##### Available metrics
+
+* Also a couple of metrics have been added:
+
+Metric|Name|Description
+---|---|---
+Number of Cache Misses|`cache_misses`|The number of cache misses
+Number of Cache Hits|`cache_hits`|The number of cache hits
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply content/guides/integrations/memcached/Go.md]: Roger that, thanks!

File: content/guides/integrations/memcached/Go.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+all now take in a context.Context object as the first argument, to allow for trace propagation.
+
+##### Available metrics
+
+* Also a couple of metrics have been added:
+
+Metric|Name|Description
+---|---|---
+Number of Cache Misses|`cache_misses`|The number of cache misses
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply content/guides/integrations/memcached/Go.md]: For sure, thanks!

File: content/guides/integrations/memcached/Go.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+[Memcached](https://memcached.org) is one of the most used server caching and scaling technologies.
+
+It was created by [Brad Fitzpatrick](https://en.wikipedia.org/wiki/Brad_Fitzpatrick) in 2003 as a 
+solution to scale his social media product [Live Journal](https://www.livejournal.com/)
+
+#### Features
+
+##### Trace updates
+OpenCensus Go was used to instrument Brad's original Memcache client, with 2 major changes:
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply content/guides/integrations/memcached/Go.md]: Roger that, thanks!

**Comment by odeke-em** [comment]
Thanks @songy23, I've addressed your feedback, PTAL.

File: content/guides/integrations/memcached/Go.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (songy23) [reply content/guides/integrations/memcached/Go.md]: Yep got it.

**Comment by odeke-em** [comment]
Awesome, thanks @songy23!

**Pull request merged**